tensorflow/core/platform/hadoop/hadoop_file_system.cc

```diff
@@ -147,6 +147,10 @@ Status HadoopFileSystem::Connect(StringPiece fname, hdfsFS* fs) {
   hdfsBuilder* builder = hdfs_->hdfsNewBuilder();
   if (scheme == "file") {
     hdfs_->hdfsBuilderSetNameNode(builder, nullptr);
+  } else if (scheme == "viewfs") {
+    // The default NameNode configuration will be used (from the XML configuration files). See:
+    // https://github.com/tensorflow/tensorflow/blob/v1.0.0/third_party/hadoop/hdfs.h#L259
+    hdfs_->hdfsBuilderSetNameNode(builder, "default");
```

Comment:
Right, but that only works if `fs.defaultFS` is set, right? There's no way to refer to another cluster here, and it'll always use the default one.

Suppose `fs.defaultFS` is set to `viewfs://cluster4`. If someone refers to a file: `viewfs:///foo/bar`, that'll point to `viewfs://cluster4/foo/bar`. With this change, if someone uses `viewfs://cluster5/foo/bar`, that'll also point to `viewfs://cluster4/foo/bar` because cluster4 is the defaultFS.

I couldn't figure out how to pass the viewfs cluster to the HDFS C API. If we can't, how about checking that nn is an empty string, otherwise return an error? That way viewfs is only supported as a defaultFS.